ndnfs/driver/driver.go

```diff
@@ -0,0 +1,360 @@
+	var mnt string
+	nfsMap, err := d.ListVolumes()
+	if err != nil || r.Name == "" {
+		log.Info("Volume with name ", r.Name, " not found")
+		return volume.Response{}
+	}
+
+	for k, v := range nfsMap {
+		if k == r.Name {
+			mnt = v
+			break
+		}
+	}
+	if mnt == "" {
+		return volume.Response{}
+	}
+
+	log.Debug("Device mountpoint is: ", mnt)
+	return volume.Response{Volume: &volume.Volume{
+		Name: r.Name, Mountpoint: mnt}}
+}
+
+func (d NdnfsDriver) List(r volume.Request) volume.Response {
+	log.Debug(DN, "List volume")
+	vmap, err := d.ListVolumes()
+	if err != nil {
+		log.Info("Failed to retrieve volume list", err)
+		return volume.Response{Err: err.Error()}
+	}
+	log.Debug(DN, "Nedge response: ", vmap)
+	var vols []*volume.Volume
+	for name, mnt := range vmap {
+		if name != "" {
+			vols = append(vols, &volume.Volume{Name: name, Mountpoint: mnt})
+		}
+	}
+	return volume.Response{Volumes: vols}
+}
+
+func (d NdnfsDriver) Mount(r volume.MountRequest) volume.Response {
+	log.Info(DN, "Mount volume: ", r.Name)
+	d.Mutex.Lock()
+	defer d.Mutex.Unlock()
+	var err error
+	var mnt string
+
+	nfs := fmt.Sprintf("%s:/%s", d.Config.Nedgedata, r.Name)
+	mnt = filepath.Join(d.Config.Mountpoint, r.Name)
+	log.Debug(DN, "Creating mountpoint folder: ", mnt)
+	if out, err := exec.Command("mkdir", "-p", mnt).CombinedOutput(); err != nil {
+	    log.Info("Error running mkdir command: ", err, "{", string(out), "}")
+	}
+	log.Debug(DN, "Mounting Volume ", r.Name)
+	args := []string{"-t", "nfs", nfs, mnt}
+	if out, err := exec.Command("mount", args...).CombinedOutput(); err != nil {
+		log.Error("Error running mount command: ", err, "{", string(out), "}")
+		err = errors.New(fmt.Sprintf("%s: %s", err, out))
+		if err != nil {
+			return volume.Response{Err: err.Error()}
+		}
+	}
+	return volume.Response{Mountpoint: mnt}
+}
+
+func (d NdnfsDriver) Path(r volume.Request) volume.Response {
+	log.Info(DN, "Path volume: ", r.Name, " Options: ", r.Options)
+	mnt := fmt.Sprintf("%s%s", d.Config.Mountpoint, r.Name)
+	return volume.Response{Mountpoint: mnt}
+}
+
+func (d NdnfsDriver) Remove(r volume.Request) volume.Response {
+	log.Info(DN, "Remove volume: ", r.Name, " Options: ", r.Options)
+	d.Mutex.Lock()
+	defer d.Mutex.Unlock()
+	nfsList, err := d.GetNfsList()
+	if err != nil {
+		log.Info("Error getting nfs list", err)
+	}
+	var path, service string
+	for i := range(nfsList) {
+		if strings.Contains(nfsList[i], r.Name) {
+			path = nfsList[i]
+		}
+	}
+	if os.Getenv("CCOW_SVCNAME") != "" {
+		service = os.Getenv("CCOW_SVCNAME")
+	} else {
+		service = d.Config.Servicename
+	}
+	data := make(map[string]interface{})
+	data["serve"] = path
+	url := fmt.Sprintf("service/%s/serve", service)
+	_, err = d.Request("DELETE", url, data)
+	if err != nil {
+		log.Info("Error while handling request", err)
+	}
+
+	parts := strings.Split(path, "/")
+	url = fmt.Sprintf("clusters/%s/tenants/%s/buckets/%s", parts[0], parts[1], parts[2])
```

Comment:
It will panics in case of err == nil or nsfList is empty with error "panic: runtime error: index out of range"